### ISSUE TITLE
csharp: Shift+Alt+F not honouring "editor.tabSize": 2

### ISSUE THREAD

**Issue by jrieken** [OmniSharp/omnisharp-vscode#24]
_From @pflannery on January 27, 2016 12:50_

v0.10.6
Executing Shift+Alt+F in a csharp file always defaults to a tab size of 4 regardless of my editor.tabSize preference.
But pressing tab in a csharp file does honour my tab size preference.

Also Shift+Alt+F works in javascript files.

_Copied from original issue: Microsoft/vscode#2439_

**Comment by pflannery** [comment]
Thanks this has led me to find a work around for now. Relates with [/OmniSharp/omnisharp-roslyn/issues/366](/OmniSharp/omnisharp-roslyn/issues/366)

So adding an `omnisharp.json` file to the root of the working folder in vs code lets me control some basic formatting.

omnisharp.json content:
```json
{
  "FormattingOptions": {
    "useTabs": false,
    "tabSize": 2,
    "indentationSize": 2,
    "smartIndent": 2,
    "newLine": "\n"
  }
}
```

Do you know if there is way I can also specify the CSharpFormattingOptions too?

**Comment by TJSoftware** [comment]
@jrieken Thank you for keeping my issue current with the multiple submissions in different repos.  

I tested this workaround and I am unable to get it to work for me.  I tried placing the `omisharp.json` file in the root as well as in the `.vscode` directory.  Here is a screenshot of my test.  My cs class has 4 spaces and the tab size is setup to be 2.  I removed the custom bindings to match Visual Studio so I only need to press `Shift + Alt + F`.  Here is a screenshot of my current test.

![image](https://cloud.githubusercontent.com/assets/4823412/12850413/d1a7f132-cbe1-11e5-846f-9d5e2cf9f300.png)

**Comment by pflannery** [comment]
@TJSoftware one extra thing that has to be done is to restart vs code after creating the omnisharp.json file.

**Comment by TJSoftware** [comment]
@pflannery Yes, I had done that, just in case I did it again.  I still have the same results.  I tested this on both my host and vm machines.  They are both Windows 10 pro.  I rebooted the vm just for fun, still no change.

**Comment by pflannery** [comment]
@TJSoftware looking again at your screen shot it might be that the "editor.tabSize" is set to 4 in your user\workspace settings. In your screen shot I can see 4 spaces from left but only two spaces once inside the c# class block.
Add some properties and methods to the class and test the format command. To remove the initial first four space issue then set  "editor.tabSize" to 2. 

![image](https://cloud.githubusercontent.com/assets/1727302/12864842/3351a678-ccd5-11e5-9335-fa44fc41b9a9.png)

**Comment by TJSoftware** [comment]
@pflannery I actually tried both the workspace and the user settings before.  My original submission is [here](https://github.com/Microsoft/vscode/issues/2554).  I double checked to make sure my current settings have this.  Here is another screenshot with relevant settings.  I am still seeing issues with the format in `C#`.  I also mentioned that I am seeing the same issue in `XML`, a non MS person said that was not supported, however I was using a [StackOverflow post](http://stackoverflow.com/questions/29973357/how-do-you-format-code-in-visual-studio-code-vscode/30031398#30031398) from @jrieken  as a reference.  I am not sure if it is supported or not.  The tabs format as expected for `JS` and `JSON`.

![image](https://cloud.githubusercontent.com/assets/4823412/12890536/54424ece-ce41-11e5-84da-6e3c8613d6d4.png)

**Comment by TJSoftware** [comment]
I just noticed an update notification on VSCode.  I ran that and I still see the issues.

I also noticed something else strange.  The version info shows that I am on node.4.1.1, if I look at cmd prompt I see node is 5.2.0.

![image](https://cloud.githubusercontent.com/assets/4823412/12921281/b76c5044-cf09-11e5-8e81-b7c5c526147f.png)
![image](https://cloud.githubusercontent.com/assets/4823412/12921304/ccd78c14-cf09-11e5-9932-b76eaf9776d1.png)

**Comment by TJSoftware** [comment]
I tested with the same as before in the latest insider.

![image](https://cloud.githubusercontent.com/assets/4823412/13538652/17b40322-e209-11e5-857d-9b0583739c07.png)

The results are the same, formatting works in `json` but not in `C#`.  I was mainly just playing with the insider build, I did not expect that this would be fixed as it is still open.

Node still shows as 4.1.1

![image](https://cloud.githubusercontent.com/assets/4823412/13538760/9c26e07a-e209-11e5-89a8-e294b6f68e1c.png)


P.S. thanks for sharing the insider builds :+1:
